libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/store/AWSInstanceMetadataFetcher.java

```diff
@@ -0,0 +1,263 @@
+/*
+ *  Copyright 2020 Verizon Media
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.yahoo.athenz.common.server.store;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.yahoo.athenz.zms.ResourceException;
+import com.yahoo.rdl.JSON;
+import com.yahoo.rdl.Struct;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+public class AWSInstanceMetadataFetcher implements Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AWSInstanceMetadataFetcher.class);
+
+    String awsRole = null;
+
+    String awsRegion;
+    private HttpClient httpClient;
+
+    public AWSInstanceMetadataFetcher() {
+        httpClient = new HttpClient();
```

Comment:
New class extracted from CloudStore. Gets the AWS Instance metadata for AWSCredentialsRefresher that provide credentials for CloudStore and for the S3ClientFactory (that creates S3Clients for the S3ChangeLogStore)